overlord/snapstate/aliasesv2.go

```diff
@@ -342,3 +342,35 @@ func checkAliasesConflicts(st *state.State, snapName string, candAutoDisabled bo
 	}
 	return nil, nil
 }
+
+// disableAliases returns newAliases corresponding to the disabling of curAliases, for manual aliases that means removed.
+func disableAliases(curAliases map[string]*AliasTarget) (newAliases map[string]*AliasTarget) {
+	newAliases = make(map[string]*AliasTarget, len(curAliases))
+	for alias, curTarget := range curAliases {
+		if curTarget.Auto != "" {
+			newAliases[alias] = &AliasTarget{Auto: curTarget.Auto}
+		}
+	}
+	return newAliases
+}
+
+// pruneAutoAliases returns newAliases by dropping the automatic aliases autoAliases from curAliases, used as the task prune-auto-aliases to handle transfers of automatic aliases in a refresh.
```

Comment:
I think there is a bit of confusion here,  this is not for "prefer" and friends, "prefer" will indeed remove also manual aliases . This is for a corner case that we need to support though (in the old world we were using the task for snap alias --reset for this), this is about automatic aliases that go from being exposed from snap A to being exposed by another snap B (to be realistic this would probably involve related snaps or some snap being split for some reason)